test/acoustic.jl

```diff
@@ -0,0 +1,60 @@
+using FEniCS
+using PyCall
+
+@pyimport fenics
+c = 5000
+#problem variables, have been scaled down for faster test solution
+dt = 0.00004;
+t = 0;
+T = 0.0004;
+
+mesh = RectangleMesh(Point([-2., -2.]),Point([2., 2.]),40,40)
+V = FunctionSpace(mesh,"Lagrange",1)
+
+# Previous and current solution
+u1= interpolate(Constant(0.0), V)
+u0= interpolate(Constant(0.0), V)
+
+# Variational problem at each time
+u = TrialFunction(V)
+v = TestFunction(V)
+
+a = u*v*dx + dt*dt*c*c*inner(grad(u), grad(v))*dx
+L = 2*u1*v*dx-u0*v*dx
+
+delta = Expression(fenics.Expression("sin(c*10*t)",degree=2,c=c,t=t))
+
+#Define boundary conditions
+top ="on_boundary && near(x[1], 2)"
+BCT = DirichletBC(V,Constant(0.0),top)
+
+down ="on_boundary && near(x[1], -2)"
+BCD = DirichletBC(V,Constant(0.0),down)
+
+left = "on_boundary && near(x[0],-2)"
+BCL = DirichletBC(V,delta,left)
+
+right = "on_boundary && near(x[0],2)"
+BCR = DirichletBC(V,Constant(0.0),right)
+
+#bcs = [BCL.pyobject,BCD.pyobject,BCT.pyobject,BCR.pyobject]
+bcs = [BCL.pyobject,BCD.pyobject,BCT.pyobject]
+
+#bc = DirichletBC(V, 0, "on_boundary")
+A, b = assemble_system(a, L, bcs)
+
+u=FEniCS.Function(V)
+
+while t <= T
+    A, b = assemble_system(a, L, bcs)
```

Comment:
You have to re-assemble every step?

Reply:
yes, the boundary  condition is time_dependent

Reply:
I see (that's precisely why our solution splits BCs out zzz :) )

Reply:
turns out so does FEniCS to a certaint extent, if you modify the way you enter the boundary conditions. I'll upload the modified code to allow this after FEniCS'18 I guess. A very basic timing estimate shows it's slightly faster as well.

Reply:
yeah that's good to know, because it seems insane to re-create the stiffness matrix every time step just for the BCs.